src/crypto/PSAOperationalKeystore.cpp

```diff
@@ -0,0 +1,213 @@
+/*
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "PSAOperationalKeystore.h"
+
+#include <lib/support/CHIPMem.h>
+
+#include <psa/crypto.h>
+
+namespace chip {
+namespace Crypto {
+
+PSAOperationalKeystore::PersistentP256Keypair::PersistentP256Keypair(FabricIndex fabricIndex)
+{
+    toPSAContext(mKeypair).key_id = MakeOperationalKeyId(fabricIndex);
+    mInitialized                  = true;
+}
+
+PSAOperationalKeystore::PersistentP256Keypair::~PersistentP256Keypair()
+{
+    // This class requires explicit control of the key lifetime. Therefore, clear the key ID
+    // to prevent it from being destroyed by the base class destructor.
+    toPSAContext(mKeypair).key_id = 0;
+}
+
+inline psa_key_id_t PSAOperationalKeystore::PersistentP256Keypair::GetKeyId() const
+{
+    return toConstPSAContext(mKeypair).key_id;
+}
+
+bool PSAOperationalKeystore::PersistentP256Keypair::Exists() const
+{
+    psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
+    psa_status_t status             = psa_get_key_attributes(GetKeyId(), &attributes);
+
+    psa_reset_key_attributes(&attributes);
+
+    return status == PSA_SUCCESS;
+}
+
+CHIP_ERROR PSAOperationalKeystore::PersistentP256Keypair::Generate()
+{
+    CHIP_ERROR error                = CHIP_NO_ERROR;
+    psa_status_t status             = PSA_SUCCESS;
+    psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
+    psa_key_id_t keyId              = 0;
+    size_t publicKeyLength;
+
+    Destroy();
+
+    // Type based on ECC with the elliptic curve SECP256r1 -> PSA_ECC_FAMILY_SECP_R1
+    psa_set_key_type(&attributes, PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1));
+    psa_set_key_bits(&attributes, kP256_PrivateKey_Length * 8);
+    psa_set_key_algorithm(&attributes, PSA_ALG_ECDSA(PSA_ALG_SHA_256));
+    psa_set_key_usage_flags(&attributes, PSA_KEY_USAGE_EXPORT | PSA_KEY_USAGE_SIGN_MESSAGE);
```

Comment:
`PSA_KEY_USAGE_EXPORT ` is not required to export the public part of the key (see https://armmbed.github.io/mbed-crypto/html/api/keys/policy.html#c.PSA_KEY_USAGE_EXPORT )

Reply:
Good catch, I needed export only for the testing and forgot to remove it.